### DISCUSSION TITLE
Any really good tutorials or examples for iOS

### DISCUSSION THREAD

**Discussion by ThatRendle** [AvaloniaUI/Avalonia#16663] [General]
I'm building something that has native apps for Windows, macOS, Linux, iOS and Android. I'm really happy with the desktop apps after a couple of tries, and definitely loving ReactiveUI, so I'd like to use Avalonia for the mobile apps too, starting with iOS because I have an iPhone.

I can get a basic app running OK on my M1 mac Mini, but I need to implement some other stuff:

- A `Share Extension` for iOS so my app shows up in the Share To list
  - And whatever the Android equivalent of that is
- Notifications

If anyone knows of some good resources to help me get my head around this I'd very much appreciate recommendations. Thanks!

**Comment by MikeCodesDotNET** [comment]
Hey @markrendle, great to see you're trying Avalonia ❤️

I think (but maybe @emmauss or @maxkatz6 can confirm) that the Sharing API should probably come from MAUI Essentials, as you'd want to use the native views. 

As for Notifications, this will probably require tweaking your AppDelegate. I'm not sure of any examples of this, but its a good request for the team ;)

**Comment by thevortexcloud** [comment]
Depending on what you mean by "notifications", Avalonia has built in support for them. They are just not native notifications, which means they only show up inside/ontop of Avalonia views.

As for sharing, MAUI/Xamarin essentials can do that for you.

https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/data/share?view=net-maui-8.0

There is examples here on how to set up MAUI essentials with Avaloinia.

https://github.com/AvaloniaUI/AvaloniaMauiHybrid


Do keep in mind: MAUI does not support Linux. As such you will have to write your own implementations for a lot of stuff there. Luckily it's generally _very_ easy to do so (often easier than Windows for most of the stuff people use Xamarin/MAUI essentials for). EG https://github.com/AvaloniaUI/Avalonia/discussions/15914

**Reply by ThatRendle** [reply]
The MAUI share integration is to show the Share UI in your app to send something, not to register your app's Share Extension with iOS.

**Reply by thevortexcloud** [reply]
Ah, I am not sure what the process for that is on iOS. On Android you have to register things in the manifest file as well as set up the activity in a specific way.

**Comment by ThatRendle** [comment]
I know very little about how Xcode projects work and stuff, but is it possible to have an Avalonia app as the "main" app and write a Share Extension and background fetch system in Swift and have that all build as one package?

**Reply by emmauss** [reply]
Yes. We don't have any api for that, but avalonia runs on existing dotnet platforms, including iOS. You are free to use any platform feature that dotnet provides.
